### PR TITLE
feat: create comment

### PR DESCRIPTION
**Pull request by idowuseyi**
**What does the PR do?**

This PR is about creating a comment

**How should this be manually tested?**

1. Login to a user account, and get the access token
2. Send a POST request to `/api/v1/comments`

Successful Response

On a successful comment creation, the API should return a 201 SUCCESS status code.

```json
{
      "status": "success",
      "message": "Comment created successfully",
      "data": {
            "blog_id": "a9f4b3b0-4b3b-4b3b-4b3b-4b3b4b3b4b3b",
            "fullUser": "John Doe",
            "content":"Comment content as supplied by the user",
      }
}
```

Error Handling:

When the current user does not have permission or is not found:

```json
{
          "status": "error",
          "message": "User nof found",
          "status_code": "404",
 }
```

Error During coment creation

If the comment cretion is not successful:

```json
{
  "status": "Error",
  "message": "Error creating comment",
  "status_code": 500
}
```

 **Checklist of what has been done**
- [X ] Define API endpoint to create comment.
- [X ]  Validate and handle authentication token.
- [ X]  Implement logic to create a comment.
- [ X]  Handle success and error responses.
- [ X]  Test Successful response

**Test Screenshot**

Reference Issue: [Link to Issue](https://github.com/hngprojects/hng_boilerplate_nestjs/issues/151)

### CODE REVIEW

File: src/modules/comments/comments.controller.ts
Comment (Heba-WebDev) [inline src/modules/comments/comments.controller.ts:1]: Please remove this comment // @UseGuards(AuthGuard)

Comment (Heba-WebDev) [reply src/modules/comments/comments.controller.ts:1]: Where should comments be? Under what exactly? Is it for blogs? Products? The answer will cause modifications to the controller and service.

Comment (idowuseyi) [reply src/modules/comments/comments.controller.ts:1]: It's all a mistake.

**Comment by idowuseyi** [comment]
A lot of things to fix.

**Pull request closed without merging**